mst/dblink-extension.md

```diff
@@ -0,0 +1,142 @@
+
+# Use the PostgreSQL `dblink` extension
+
+`dblink` is a [PostgreSQLextension](https://www.postgresql.org/docs/current/dblink.html)
+that allows you to connect to other PostgreSQL databases and to run arbitrary queries.
+
+With [Foreign Data Wrappers](https://www.postgresql.org/docs/current/postgres-fdw.html) (FDW)
+you can define a remote `foreign server` in order to access its data. The
+database connection details such as hostnames are kept in a single place, and you
+only need to create a `user mapping` to store remote connections credentials.
+
+## Before you begin
+
+To create a `dblink` foreign data wrapper you need the following information
+about the PostgreSQL remote server:
+
+*   `HOSTNAME`: The remote database hostname
+*   `PORT`: The remote database port
+*   `USER`: The remote database user to connect. The default user is `tsdbadmin`.
+*   `PASSWORD`: The remote database password for the
+  `USER`
+*   `DATABASE_NAME`: The remote database name. The default database name is `defaultdb`.
+
+<highlight type="note">
+The details are available in the [MST portal](https://portal.managed.timescale.com)
+service `Overview` tab or use the `avn service get` command if you installed
+Aiven client.
+</highlight>
+
+<procedure>
+
+### Enable `dblink` extension on MST for PostgreSQL
+
+To enable the `dblink` extension on an MST PostgreSQL service:
+
+1.  Connect to the database as the `tsdbadmin` user:
+
+```bash
+    psql -x "postgres://tsdbadmin:<PASSWORD>@<HOSTNAME>:<PORT>/defaultdb?sslmode=require"
+```
+
+1.  Create the `dblink` extension
+
+```sql
+    CREATE EXTENSION dblink;
+```
+
+1.  Create a table named `inventory`:
+
+```sql
+    CREATE TABLE inventory (id int);
+```
+
+1.  Insert data into the `inventory` table:
+
+```sql
+    INSERT INTO inventory (id) VALUES (100), (200), (300);
+```
+
+</procedure>
+
+<procedure>
+
+### Create a foreign data wrapper using `dblink_fdw`
+
+1.  Create a user `user1` who can access the `dblink`
+
+```sql
+    CREATE USER user1 PASSWORD 'secret1'
+```
+
+1.  Create a remote server definition named `mst_remote`, using `dblink_fdw` and
+    the connection details of the MST service.
+
+```sql
+
+    CREATE SERVER mst_remote
+        FOREIGN DATA WRAPPER dblink_fdw
+        OPTIONS (
+                 host 'HOST',
+                 dbname 'DATABASE_NAME', 
+                 port 'PORT'
+                 );
+```
+
+1.  Create a user mapping for the `user1` to automatically authenticate as the
+    `tsdbadmin` when using the   `dblink`:
+
+```sql
+
+    CREATE USER MAPPING FOR user1
+        SERVER mst_remote
+        OPTIONS (
+            user 'tsdbadmin', 
+            password 'PASSWORD'
+            );
+```
+
+1.  Enable `user1` to use the remote PostgreSQL connection `mst_remote`:
+
+```sql
+    GRANT USAGE ON FOREIGN SERVER mst_remote TO user1;
+```
+
+</procedure>
+
+### Query data using a foreign data wrapper
+
+To query a foreign data wrapper, you must be a database user with the necessary
```

Comment:
This whole section doesn't separate the concept from the task well. Can you please re-write to put all the concepts up front, then add in a procedure block with the task?

Reply:
agreed!!!